classes/Project/Project.mjs

```diff
@@ -67,33 +109,109 @@ export default class Project {
       }
     }
 
-    if (
-      !this.projectData.groups.members ||
-      this.projectData.groups.members.length === 0
-    ) {
+    const member = Object.values(this.projectData.groups).find(
```

Comment:
I think this will be built to be `projectData.contributors`. At any rate, a Project will only have one "Group" so the `groups` seems to actually be like `groupmembers`.

Reply:
Also, if the `.contributors` is a User Map, this could easily just be `const permissions = projectData.contributors[userAgent]?.permissions` and then `!permissions` means they aren't in the group.

The `action` argument is not used yet, but it could also be attempted to match with a failsafe. Then the error could just be "Not allowed" if `permissions` has length and "Not a member" if there are none.